ClangTidy warnings for `misc-use-anonymous-namespace`

```
/Users/jviotti/Projects/core/src/core/jsonschema/frame.cc:15:13: error: function 'find_anchors' declared 'static', move to anonymous namespace instead [misc-use-anonymous-namespace,-warnings-as-errors]
   15 | static auto find_anchors(const sourcemeta::core::JSON &schema,
      |             ^

/Users/jviotti/Projects/core/src/core/jsonschema/frame.cc:101:13: error: function 'find_nearest_bases' declared 'static', move to anonymous namespace instead [misc-use-anonymous-namespace,-warnings-as-errors]
  101 | static auto find_nearest_bases(
      |             ^

/Users/jviotti/Projects/core/src/core/jsonschema/frame.cc:230:13: error: function 'traverse_origin_instance_locations' declared 'static', move to anonymous namespace instead [misc-use-anonymous-namespace,-warnings-as-errors]
  230 | static auto traverse_origin_instance_locations(
      |             ^

/Users/jviotti/Projects/core/src/core/jsonschema/frame.cc:267:13: error: function 'repopulate_instance_locations' declared 'static', move to anonymous namespace instead [misc-use-anonymous-namespace,-warnings-as-errors]
  267 | static auto repopulate_instance_locations(
      |             ^

/Users/jviotti/Projects/core/src/core/jsonschema/jsonschema.cc:16:13: error: function 'id_keyword_guess' declared 'static', move to anonymous namespace instead [misc-use-anonymous-namespace,-warnings-as-errors]
   16 | static auto id_keyword_guess(const sourcemeta::core::JSON &schema)
      |             ^

/Users/jviotti/Projects/core/src/core/jsonschema/jsonschema.cc:31:13: error: function 'id_keyword' declared 'static', move to anonymous namespace instead [misc-use-anonymous-namespace,-warnings-as-errors]
   31 | static auto id_keyword(const std::string &base_dialect) -> std::string {
      |             ^

/Users/jviotti/Projects/core/src/core/jsonschema/resolver.cc:92:13: error: function 'to_lowercase' declared 'static', move to anonymous namespace instead [misc-use-anonymous-namespace,-warnings-as-errors]
   92 | static auto to_lowercase(const std::string_view input) -> std::string {
      |             ^

/Users/jviotti/Projects/core/src/core/uri/uri.cc:16:13: error: function 'uri_normalize' declared 'static', move to anonymous namespace instead [misc-use-anonymous-namespace,-warnings-as-errors]
   16 | static auto uri_normalize(UriUriA *uri) -> void {
      |             ^

/Users/jviotti/Projects/core/src/core/uri/uri.cc:22:13: error: function 'uri_to_string' declared 'static', move to anonymous namespace instead [misc-use-anonymous-namespace,-warnings-as-errors]
   22 | static auto uri_to_string(const UriUriA *const uri) -> std::string {
      |             ^

/Users/jviotti/Projects/core/src/core/uri/uri.cc:39:13: error: function 'uri_text_range' declared 'static', move to anonymous namespace instead [misc-use-anonymous-namespace,-warnings-as-errors]
   39 | static auto uri_text_range(const UriTextRangeA *const range)
      |             ^

/Users/jviotti/Projects/core/src/core/uri/uri.cc:50:13: error: function 'uri_parse' declared 'static', move to anonymous namespace instead [misc-use-anonymous-namespace,-warnings-as-errors]
   50 | static auto uri_parse(const std::string &data, UriUriA *uri) -> void {
      |             ^

/Users/jviotti/Projects/core/src/core/uri/uri.cc:70:13: error: function 'canonicalize_path' declared 'static', move to anonymous namespace instead [misc-use-anonymous-namespace,-warnings-as-errors]
   70 | static auto canonicalize_path(const std::string &path)
      |             ^

/Users/jviotti/Projects/core/src/core/yaml/yaml.cc:12:13: error: function 'yaml_node_to_json' declared 'static', move to anonymous namespace instead [misc-use-anonymous-namespace,-warnings-as-errors]
   12 | static auto yaml_node_to_json(yaml_node_t *const node,
      |             ^

/Users/jviotti/Projects/core/src/core/yaml/yaml.cc:84:13: error: function 'internal_parse_json' declared 'static', move to anonymous namespace instead [misc-use-anonymous-namespace,-warnings-as-errors]
   84 | static auto internal_parse_json(yaml_parser_t *parser)
      |             ^

```

--- a/src/core/jsonschema/frame.cc
+++ b/src/core/jsonschema/frame.cc
@@ -100,7 +100,7 @@ auto find_anchors(const sourcemeta::core::JSON &schema,
   return result;
 }
 
-static auto find_nearest_bases(
+auto find_nearest_bases(
     const std::map<sourcemeta::core::Pointer,
                    std::vector<sourcemeta::core::JSON::String>> &bases,
     const sourcemeta::core::Pointer &pointer,
@@ -120,9 +120,8 @@ static auto find_nearest_bases(
   return {{}, sourcemeta::core::empty_pointer};
 }
 
-} // namespace
 
-static auto find_every_base(
+auto find_every_base(
     const std::map<sourcemeta::core::Pointer,
                    std::vector<sourcemeta::core::JSON::String>> &bases,
     const sourcemeta::core::Pointer &pointer)
@@ -149,7 +148,7 @@ static auto find_every_base(
   return result;
 }
 
-static auto ref_overrides_adjacent_keywords(
+auto ref_overrides_adjacent_keywords(
     const sourcemeta::core::JSON::String &base_dialect) -> bool {
   // In older drafts, the presence of `$ref` would override any sibling
   // keywords
@@ -165,7 +164,7 @@ static auto ref_overrides_adjacent_keywords(
          base_dialect == "http://json-schema.org/draft-03/hyper-schema#";
 }
 
-static auto
+auto
 supports_id_anchors(const sourcemeta::core::JSON::String &base_dialect)
     -> bool {
   return base_dialect == "http://json-schema.org/draft-07/schema#" ||
@@ -176,7 +175,7 @@ supports_id_anchors(const sourcemeta::core::JSON::String &base_dialect)
          base_dialect == "http://json-schema.org/draft-04/hyper-schema#";
 }
 
-static auto fragment_string(const sourcemeta::core::URI &uri)
+auto fragment_string(const sourcemeta::core::URI &uri)
     -> std::optional<sourcemeta::core::JSON::String> {
   const auto fragment{uri.fragment()};
   if (fragment.has_value()) {
@@ -187,14 +186,14 @@ static auto fragment_string(const sourcemeta::core::URI &uri)
 }
 
 [[noreturn]]
-static auto throw_already_exists(const sourcemeta::core::JSON::String &uri)
+auto throw_already_exists(const sourcemeta::core::JSON::String &uri)
     -> void {
   std::ostringstream error;
   error << "Schema identifier already exists: " << uri;
   throw sourcemeta::core::SchemaError(error.str());
 }
 
-static auto
+auto
 store(sourcemeta::core::SchemaFrame::Locations &frame,
       sourcemeta::core::SchemaFrame::Instances &instances,
       const sourcemeta::core::SchemaReferenceType type,
@@ -231,8 +230,6 @@ struct InternalEntry {
   const std::optional<sourcemeta::core::JSON::String> id;
 };
 
-namespace {
-
 auto traverse_origin_instance_locations(
     const sourcemeta::core::SchemaFrame &frame,
     const sourcemeta::core::SchemaFrame::Instances &instances,
@@ -263,16 +260,12 @@ auto traverse_origin_instance_locations(
   }
 }
 
-} // namespace
-
 struct CacheSubschema {
   const sourcemeta::core::PointerTemplate instance_location;
   const sourcemeta::core::PointerTemplate relative_instance_location;
   const bool orphan;
   const std::optional<sourcemeta::core::Pointer> parent;
 };
-
-namespace {
 
 auto repopulate_instance_locations(
     const sourcemeta::core::SchemaFrame &frame,

--- a/src/core/jsonschema/frame.cc
+++ b/src/core/jsonschema/frame.cc
@@ -12,6 +12,8 @@
 
 enum class AnchorType : std::uint8_t { Static, Dynamic, All };
 
+namespace {
+
 static auto find_anchors(const sourcemeta::core::JSON &schema,
                          const sourcemeta::core::Vocabularies &vocabularies)
     -> std::map<sourcemeta::core::JSON::String, AnchorType> {
@@ -117,6 +119,8 @@ static auto find_nearest_bases(
 
   return {{}, sourcemeta::core::empty_pointer};
 }
+
+} // namespace
 
 static auto find_every_base(
     const std::map<sourcemeta::core::Pointer,
@@ -227,6 +231,8 @@ struct InternalEntry {
   const std::optional<sourcemeta::core::JSON::String> id;
 };
 
+namespace {
+
 static auto traverse_origin_instance_locations(
     const sourcemeta::core::SchemaFrame &frame,
     const sourcemeta::core::SchemaFrame::Instances &instances,
@@ -257,12 +263,16 @@ static auto traverse_origin_instance_locations(
   }
 }
 
+} // namespace
+
 struct CacheSubschema {
   const sourcemeta::core::PointerTemplate instance_location;
   const sourcemeta::core::PointerTemplate relative_instance_location;
   const bool orphan;
   const std::optional<sourcemeta::core::Pointer> parent;
 };
+
+namespace {
 
 static auto repopulate_instance_locations(
     const sourcemeta::core::SchemaFrame &frame,
@@ -313,6 +323,8 @@ static auto repopulate_instance_locations(
     }
   }
 }
+
+} // namespace
 
 namespace sourcemeta::core {
 

--- a/src/core/jsonschema/frame.cc
+++ b/src/core/jsonschema/frame.cc
@@ -120,7 +120,6 @@ auto find_nearest_bases(
   return {{}, sourcemeta::core::empty_pointer};
 }
 
-
 auto find_every_base(
     const std::map<sourcemeta::core::Pointer,
                    std::vector<sourcemeta::core::JSON::String>> &bases,
@@ -164,8 +163,7 @@ auto ref_overrides_adjacent_keywords(
          base_dialect == "http://json-schema.org/draft-03/hyper-schema#";
 }
 
-auto
-supports_id_anchors(const sourcemeta::core::JSON::String &base_dialect)
+auto supports_id_anchors(const sourcemeta::core::JSON::String &base_dialect)
     -> bool {
   return base_dialect == "http://json-schema.org/draft-07/schema#" ||
          base_dialect == "http://json-schema.org/draft-07/hyper-schema#" ||
@@ -186,28 +184,27 @@ auto fragment_string(const sourcemeta::core::URI &uri)
 }
 
 [[noreturn]]
-auto throw_already_exists(const sourcemeta::core::JSON::String &uri)
-    -> void {
+auto throw_already_exists(const sourcemeta::core::JSON::String &uri) -> void {
   std::ostringstream error;
   error << "Schema identifier already exists: " << uri;
   throw sourcemeta::core::SchemaError(error.str());
 }
 
-auto
-store(sourcemeta::core::SchemaFrame::Locations &frame,
-      sourcemeta::core::SchemaFrame::Instances &instances,
-      const sourcemeta::core::SchemaReferenceType type,
-      const sourcemeta::core::SchemaFrame::LocationType entry_type,
-      const sourcemeta::core::JSON::String &uri,
-      const std::optional<sourcemeta::core::JSON::String> &root_id,
-      const sourcemeta::core::JSON::String &base_id,
-      const sourcemeta::core::Pointer &pointer_from_root,
-      const sourcemeta::core::Pointer &pointer_from_base,
-      const sourcemeta::core::JSON::String &dialect,
-      const sourcemeta::core::JSON::String &base_dialect,
-      const std::vector<sourcemeta::core::PointerTemplate> &instance_locations,
-      const std::optional<sourcemeta::core::Pointer> &parent,
-      const bool ignore_if_present = false) -> void {
+auto store(
+    sourcemeta::core::SchemaFrame::Locations &frame,
+    sourcemeta::core::SchemaFrame::Instances &instances,
+    const sourcemeta::core::SchemaReferenceType type,
+    const sourcemeta::core::SchemaFrame::LocationType entry_type,
+    const sourcemeta::core::JSON::String &uri,
+    const std::optional<sourcemeta::core::JSON::String> &root_id,
+    const sourcemeta::core::JSON::String &base_id,
+    const sourcemeta::core::Pointer &pointer_from_root,
+    const sourcemeta::core::Pointer &pointer_from_base,
+    const sourcemeta::core::JSON::String &dialect,
+    const sourcemeta::core::JSON::String &base_dialect,
+    const std::vector<sourcemeta::core::PointerTemplate> &instance_locations,
+    const std::optional<sourcemeta::core::Pointer> &parent,
+    const bool ignore_if_present = false) -> void {
   assert(std::set<sourcemeta::core::PointerTemplate>(
              instance_locations.cbegin(), instance_locations.cend())
              .size() == instance_locations.size());

--- a/src/core/jsonschema/frame.cc
+++ b/src/core/jsonschema/frame.cc
@@ -14,8 +14,8 @@ enum class AnchorType : std::uint8_t { Static, Dynamic, All };
 
 namespace {
 
-static auto find_anchors(const sourcemeta::core::JSON &schema,
-                         const sourcemeta::core::Vocabularies &vocabularies)
+auto find_anchors(const sourcemeta::core::JSON &schema,
+                  const sourcemeta::core::Vocabularies &vocabularies)
     -> std::map<sourcemeta::core::JSON::String, AnchorType> {
   std::map<sourcemeta::core::JSON::String, AnchorType> result;
 
@@ -233,7 +233,7 @@ struct InternalEntry {
 
 namespace {
 
-static auto traverse_origin_instance_locations(
+auto traverse_origin_instance_locations(
     const sourcemeta::core::SchemaFrame &frame,
     const sourcemeta::core::SchemaFrame::Instances &instances,
     const sourcemeta::core::Pointer &current,
@@ -274,7 +274,7 @@ struct CacheSubschema {
 
 namespace {
 
-static auto repopulate_instance_locations(
+auto repopulate_instance_locations(
     const sourcemeta::core::SchemaFrame &frame,
     const sourcemeta::core::SchemaFrame::Instances &instances,
     const std::map<sourcemeta::core::Pointer, CacheSubschema> &cache,

--- a/src/core/jsonschema/jsonschema.cc
+++ b/src/core/jsonschema/jsonschema.cc
@@ -15,7 +15,7 @@ auto sourcemeta::core::is_schema(const sourcemeta::core::JSON &schema) -> bool {
 
 namespace {
 
-static auto id_keyword_guess(const sourcemeta::core::JSON &schema)
+auto id_keyword_guess(const sourcemeta::core::JSON &schema)
     -> std::optional<std::string> {
   if (schema.defines("$id") && schema.at("$id").is_string()) {
     if (!schema.defines("id") ||

--- a/src/core/jsonschema/jsonschema.cc
+++ b/src/core/jsonschema/jsonschema.cc
@@ -13,6 +13,8 @@ auto sourcemeta::core::is_schema(const sourcemeta::core::JSON &schema) -> bool {
   return schema.is_object() || schema.is_boolean();
 }
 
+namespace {
+
 static auto id_keyword_guess(const sourcemeta::core::JSON &schema)
     -> std::optional<std::string> {
   if (schema.defines("$id") && schema.at("$id").is_string()) {
@@ -54,6 +56,8 @@ static auto id_keyword(const std::string &base_dialect) -> std::string {
   error << "Unrecognized base dialect: " << base_dialect;
   throw sourcemeta::core::SchemaError(error.str());
 }
+
+} // namespace
 
 auto sourcemeta::core::identify(
     const sourcemeta::core::JSON &schema, const SchemaResolver &resolver,

--- a/src/core/uri/uri.cc
+++ b/src/core/uri/uri.cc
@@ -14,6 +14,8 @@
 #include <utility>    // std::move
 #include <vector>     // std::vector
 
+namespace {
+
 static auto uri_normalize(UriUriA *uri) -> void {
   if (uriNormalizeSyntaxA(uri) != URI_SUCCESS) {
     throw sourcemeta::core::URIError{"Could not normalize URI"};
@@ -112,6 +114,8 @@ static auto canonicalize_path(const std::string &path)
     return std::nullopt;
   return canonical_path;
 }
+
+} // namespace
 
 namespace sourcemeta::core {
 

--- a/src/core/uri/uri.cc
+++ b/src/core/uri/uri.cc
@@ -16,14 +16,14 @@
 
 namespace {
 
-static auto uri_normalize(UriUriA *uri) -> void {
+auto uri_normalize(UriUriA *uri) -> void {
   if (uriNormalizeSyntaxA(uri) != URI_SUCCESS) {
     throw sourcemeta::core::URIError{"Could not normalize URI"};
   }
 }
 
-static auto uri_to_string(const UriUriA *const uri) -> std::string {
-  int size = 0;
+auto uri_to_string(const UriUriA *const uri) -> std::string {
+  int size;
   if (uriToStringCharsRequiredA(uri, &size) != URI_SUCCESS) {
     throw sourcemeta::core::URIError{"Could not determine URI size"};
   }
@@ -39,7 +39,7 @@ static auto uri_to_string(const UriUriA *const uri) -> std::string {
   return result;
 }
 
-static auto uri_text_range(const UriTextRangeA *const range)
+auto uri_text_range(const UriTextRangeA *const range)
     -> std::optional<std::string_view> {
   if (range->afterLast == nullptr) {
     return std::nullopt;
@@ -50,8 +50,8 @@ static auto uri_text_range(const UriTextRangeA *const range)
                               range->afterLast - range->first)};
 }
 
-static auto uri_parse(const std::string &data, UriUriA *uri) -> void {
-  const char *error_position = nullptr;
+auto uri_parse(const std::string &data, UriUriA *uri) -> void {
+  const char *error_position;
   switch (uriParseSingleUriA(uri, data.c_str(), &error_position)) {
     case URI_ERROR_SYNTAX:
       // TODO: Test the positions of this error
@@ -70,8 +70,7 @@ static auto uri_parse(const std::string &data, UriUriA *uri) -> void {
   uri_normalize(uri);
 }
 
-static auto canonicalize_path(const std::string &path)
-    -> std::optional<std::string> {
+auto canonicalize_path(const std::string &path) -> std::optional<std::string> {
   // TODO: This is a hack, as this whole function works badly for
   // relative paths with ".."
   if (path.starts_with("..")) {

--- a/src/core/yaml/yaml.cc
+++ b/src/core/yaml/yaml.cc
@@ -11,8 +11,7 @@ namespace {
 
 // TODO: Perform parsing token by token using `yaml_parser_parse`,
 // as that function also let us get line/column information on `yaml_event_t`
-static auto yaml_node_to_json(yaml_node_t *const node,
-                              yaml_document_t *const document)
+auto yaml_node_to_json(yaml_node_t *const node, yaml_document_t *const document)
     -> sourcemeta::core::JSON {
   if (!node) {
     return sourcemeta::core::JSON{nullptr};
@@ -83,8 +82,7 @@ static auto yaml_node_to_json(yaml_node_t *const node,
   }
 }
 
-static auto internal_parse_json(yaml_parser_t *parser)
-    -> sourcemeta::core::JSON {
+auto internal_parse_json(yaml_parser_t *parser) -> sourcemeta::core::JSON {
   yaml_document_t document;
   if (!yaml_parser_load(parser, &document)) {
     // TODO: Ideally we would get line/column information like for `ParseError`

--- a/src/core/yaml/yaml.cc
+++ b/src/core/yaml/yaml.cc
@@ -7,6 +7,8 @@
 #include <sstream>     // std::ostringstream, std::istringstream
 #include <string_view> // std::string_view
 
+namespace {
+
 // TODO: Perform parsing token by token using `yaml_parser_parse`,
 // as that function also let us get line/column information on `yaml_event_t`
 static auto yaml_node_to_json(yaml_node_t *const node,
@@ -104,6 +106,8 @@ static auto internal_parse_json(yaml_parser_t *parser)
     throw;
   }
 }
+
+} // namespace
 
 namespace sourcemeta::core {
 

